lib/cext/preprocess.rb

```diff
@@ -352,8 +438,13 @@ def preprocess(line)
 
 def patch(file, contents, directory)
   if patched_file = PATCHED_FILES[File.basename(file)]
-    regexp = /^#{Regexp.escape(patched_file[:gem])}\b/
-    if directory.split('/').last(2).any? { |part| part =~ regexp }
+    matched = if patched_file[:ext_dir]
+                directory.end_with?(File.join(patched_file[:gem], 'ext', patched_file[:ext_dir]))
```

Comment:
Is it like `json/ext/generator`? Is there no version next to `json` on gem install like `json-1.2.3`?

Reply:
The path looks like `lib/ruby/gems/2.3.0/gems/json-2.1.0/ext/json/ext/generator/`

Reply:
That is not what I expected, that's quite original, 2 `ext` dirs.
Could you maybe add it as a comment on this line so it's easier to edit this code and understand the requirements?